sequence/sequences/compose/arith.py

```diff
@@ -0,0 +1,74 @@
+from sequence.core.core import Sequence
+
+import numpy as np
+
+
+class Arith(Sequence):
+    """Sequences combined via arithmetic operation, e.g., addition, multiplication.
+
+    Parameters
+    ----------
+    sequences : List[Sequence]
+        List of sequences to combine.
+    operation : None, str, function, or numpy ufunc, optional, default = None = mean
+        if str, must be one of "mean", "+" (add), "*" (multiply), "max", "min"
+        if func, must be of signature (1D iterable) -> float
+        operation carried out on the sequences
+    """
+
+    def __init__(self, sequences, operation):
+        super().__init__()
+        self.sequences = sequences
+        self.operation = operation
+        self._op = self._resolve_operation(operation)
+
+    def __add__(self, other):
+        if not isinstance(other, Arith) or self._op != other._op:
+            return Arith(sequences=[*self.sequences, other], operation="+")
+        else:
+            return Arith(sequences=[*self.sequences, *other.sequences], operation="+")
+
+    def __mul__(self, other):
+        from sequence.sequences.compose.arith import Arith
+
+        if not isinstance(other, Arith) or self._op != other._op:
+            return Arith(sequences=[*self.sequences, other], operation="*")
+        else:
+            return Arith(sequences=[*self.sequences, *other.sequences], operation="*")
+
+    def _resolve_operation(self, operation):
+        """Coerce operation to a numpy.ufunc."""
+        alias_dict = {
+            None: np.mean,
+            "mean": np.mean,
+            "+": np.add,
+            "add": np.add,
+            "*": np.multiply,
+            "mult": np.multiply,
+            "multiply": np.multiply,
+            "min": np.min,
+            "max": np.max,
+        }
+
+        if operation in alias_dict.keys():
+            return alias_dict[operation]
+        else:
+            return operation
+
+    def is_finite(self) -> bool:
+        return any(sequence.is_finite() for sequence in self.sequences)
+
+    def _as_generator(self):
+        op = self._op
+        while True:
+            yield op([next(sequence) for sequence in self.sequences])
```

Comment:
this is gonna to fail if you have the sum of a finite sequence with an infinite one, or not?

and how it is defined the product of two sequences, one finite and the other infinite?

In general, we say that a sequence `a_n` is finite if there is `N > 0` such that `a_n = 0` for every `n > N`. Therefore, the product of a finite sequence with an infinite one should be finite. Is this respected in this implementation?

Reply:
oh, I see.

I didn't fully understand your model. I thought finite sequence indeed had finite length, I would expect tee generator to be finite too, in this case.

Given this explanation, I now have a strong opinion that that is how it *should* be, i.e., next should not give 0, but a generator-end.

Because otherwise, you cannot symbolically distinguish infinite sequences that have 0s, and finite sequencese; and, you cannot distinguish finite sequences with trailing 0s, from the same sequence but shorter.

Reply:
Actually I explain myself bad. I hope my explanation above is better :-)

Reply:
which explanation, is there a new one?

Reply:
I was thinking about that: https://github.com/VascoSch92/sequentium/pull/47#discussion_r1554684536

In fact, I was confusing what should be a finite sequence: a sequence with finite length or one which is eventually always 0

Reply:
so, what should our definition of finite be for the package?
(I would prefer agreeing with the one on wikipedia - finite length, rather than "becomes constant zero")